### PR TITLE
Added two new functions in Transform Extensions, translate UIView and…

### PR DESCRIPTION
**Pull request by vilapuigvila**
## Checklist

- [x] New Extension
- [ ] New Test
- [ ] Changed more than one extension, but all changes are related
- [ ] Trivial change (doesn't require changelog)

… trasnformIdentity with completion block.

### CODE REVIEW

**Comment by Khalian** [comment]
Can we add tests for these extensions?

File: Sources/UIViewExtensions.swift [hunk 1]
```diff
@@ -307,6 +309,28 @@ extension UIView {
         self.layer.transform = transform
     }
 
+    
+    /// EZSwiftExtensions
+    public func setTranslation(_ duration:TimeInterval?, _ x: CGFloat?, y: CGFloat?, completion: ((Bool) -> Void)? = nil) {
+        self.animate(duration: duration ?? UIViewTransitionDuration, animations: {
+            self.transform = .init(translationX: x ?? 0, y: y ?? 0)
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: I do not understand what these methods actually do. Can you put a comment on the top.

File: Sources/UIViewExtensions.swift [hunk 2]
```diff
@@ -307,6 +309,28 @@ extension UIView {
...
+            if let completion = completion {
+                completion(success)
+            }
+        }
+    }
+    
+    /// EZSwiftExtensions
+    public func setTranslationIdentity(_ duration:TimeInterval?, completion: ((Bool) -> Void)? = nil) {
+        self.animate(duration: duration ?? UIViewTransitionDuration, animations: {
+            self.transform = CGAffineTransform.identity
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: Is setTranslationIdentity a special case of setTranslation (as the only difference between the two is the definition of the transforms. 

Moreover are other kinds of transforms defined in CG? If so the name of the function needs to be changed to somehow incorporate the affine transform concept.

File: Sources/UIViewExtensions.swift (hunk 1 above)
Comment (vilapuigvila) [inline Sources/UIViewExtensions.swift]: OK, I'm going to do it.

File: Sources/UIViewExtensions.swift (hunk 2 above)
Comment (vilapuigvila) [inline Sources/UIViewExtensions.swift]: this function is only to back to his origin after translation, rotate, scale, etc..  I'll change the name.

File: Sources/UIViewExtensions.swift [hunk 3]
```diff
@@ -263,6 +263,8 @@ extension UIView {
     }
 }
 
+private let UIViewTransitionDuration: TimeInterval = 0.45
```
Comment (Khalian) [inline Sources/UIViewExtensions.swift]: Lets call id defaultUITransitionViewDuration instead.

Comment (vilapuigvila) [inline Sources/UIViewExtensions.swift]: I've wrote UIViewDefaultTransitionDuration, to be in context like these  UIViewAnimationDuration, UIViewAnimationSpringDamping UIViewAnimationSpringVelocity

**Comment by vilapuigvila** [comment]
I don't have too experience to adding tests, it you may help me a little, I'll be glad.
Thanks.

**Comment by lfarah** [comment]
Hey @vilapuigvila, if you want more detailed help, please hit me on Gitter.
Tests are really easy to learn and I Iearned how to write tests through EZSwiftExtensions.

**Comment by lfarah** [comment]
Also [check this out](http://www.xmcgraw.com/how-to-start-unit-testing-in-swift/)

**Comment by vilapuigvila** [comment]
Ok perfect @lfarah I'll take a look about that.

**Review by lfarah — APPROVED** [review]

File: EZSwiftExtensionsTests/UIViewTests.swift [hunk 4]
```diff
@@ -11,6 +11,51 @@ import XCTest
...
+
+    }
+    
+    func testBackToOrigin() {
+        
+        let view = self.view()
+        let aView = UIView(frame:view.frame)
+        view.addSubview(aView)
+        
+        let oldOrigin:CGPoint = CGPoint(x: aView.frame.minX, y:aView.frame.minY)
```
Comment (goktugyil) [inline EZSwiftExtensionsTests/UIViewTests.swift]: You should be getting swift lint warnings about this.

Comment (furuyan) [inline EZSwiftExtensionsTests/UIViewTests.swift]: I recommend to use code formatter.
My recommendation is [Swimat](https://github.com/Jintin/Swimat).

**Review by Esqarrouth — APPROVED** [review]

**Comment by lfarah** [comment]
Hey @vilapuigvila, this PR is approved. I just need you to fix the conflicts before I can merge

**Comment by piv199** [comment]
Do we have to include rotation, scaling too?

**Comment by vilapuigvila** [comment]
hi @lfarah I can't see conflicts, I merged master into this branch in my machine and works well, please can you help me?

**Comment by piv199** [comment]
@vilapuigvila rebase on master

**Comment by vilapuigvila** [comment]
I did, but I have no conflicts. I don't understand :-(

**Comment by piv199** [comment]
@vilapuigvila you haven't pushed to your remote

**Comment by vilapuigvila** [comment]
@piv199 which branch? mater or uiviewTranslations?

**Comment by piv199** [comment]
`vilapuigvila:uiviewTranslations`

**Comment by vilapuigvila** [comment]
@piv199 uiviewTranslations hasn't anything to push. I'm sorry but I'm a little confused..

**Comment by Khalian** [comment]
Ok now would you mind pulling the mainline in trunk and rebasing and squashing your changes on top of it ? Also with our new CHANGELOG template you have to add an entry for the same.

**Pull request closed without merging**